cicd/cicd_runner.1.sh

```diff
@@ -1,125 +1,157 @@
 docker network create testingnet
 
 # Create the Splunk container from the image but do not start it yet
-echo "Starting splunk image $version..."
-docker container create --rm --name splunk --hostname "idx-example.splunkcloud.com" --network testingnet $version
+echo "Starting splunk image ${version} as ${container}..."
+docker run -d \
+      --name $container \
+      --network testingnet \
+      --hostname "$SPLUNK_HOSTNAME" \
+      --env SPLUNK_HOME="$SPLUNK_HOME" \
+      --env SPLUNK_ETC="$SPLUNK_ETC" \
+      --env SPLUNK_START_ARGS="$SPLUNK_START_ARGS" \
+      --env SPLUNK_ENABLE_LISTEN="$SPLUNK_ENABLE_LISTEN" \
+      --env SPLUNK_ADD="$SPLUNK_ADD" \
+      --env SPLUNK_PASSWORD="$SPLUNK_PASSWORD" \
+      --user root \
+      -p 8000:8000 \
+      -p 8089:8089 \
+      $version
 
-# Copy app and configuration into Splunk container
 echo "Copying data into container..."
-echo "docker cp $CI_PROJECT_DIR/$APP_ROOT $version:$APPS_DIR/"
-docker cp $CI_PROJECT_DIR/$APP_ROOT $version:$APPS_DIR/
-echo "docker cp $CI_PROJECT_DIR/output $version:/"
-docker cp $CI_PROJECT_DIR/output $version:/
-#echo "docker cp $CI_PROJECT_DIR/cicd/inputs.conf $version:$APPS_DIR/$APP_ROOT/default/inputs.conf"
-#docker cp $CI_PROJECT_DIR/cicd/inputs.conf $version:$APPS_DIR/$APP_ROOT/default/inputs.conf
-
-# Start Splunk container
-echo "starting ${version}..."
-docker start $version
+docker exec $container bash -c "mkdir -p -m 777 /opt/splunk/etc/apps"
+docker cp $CI_PROJECT_DIR/cicd/config/passwd $container:$SPLUNK_ETC/passwd
+docker cp $CI_PROJECT_DIR/$APP_ROOT $container:$APPS_DIR/$APP_ROOT
+docker cp $CI_PROJECT_DIR/output $container:/
+# Prevent splunk from prompting for password reset
+docker exec $container bash -c "touch /opt/splunk/etc/.ui_login"
 
 # Wait for instance to be available
 # Waiting for 2 and a half minutes.
-loopCounter=30
+loopCounter=0
+health="starting"
+
+# check to see if container has a health check and if so, wait for it to be healthy
+while [[ $loopCounter -lt MAX_WAIT_SECONDS && $health =~ "starting" ]]; do
+  health=`docker ps --filter "name=${container}" --format "{{.Status}}"`
+  echo -ne "\rWaiting for Splunk to be available...$((MAX_WAIT_SECONDS - loopCounter))   "
+  ((loopCounter++))
+  sleep 1
+done
+
+# validate container is running
+health=`docker ps --filter "name=${container}" --format "{{.Status}}"`
+# if there was a problem, print some debugging information
+if [[ $health == "" ]]; then
+  echo "Health:\n${health}\n" &> Errors.txt
+  echo "--------------------------------" &> Errors.txt
+  docker ps -a &> Errors.txt
+  echo "--------------------------------" &> Errors.txt
+  docker inspect $container &> Errors.txt
+  echo "--------------------------------" &> Errors.txt
+  docker logs $container &> Errors.txt
+  echo "--------------------------------" &> Errors.txt
+  echo "Container is no longer running!"
+  echo "See Errors.txt for more information."
```

Comment:
Great idea! Makes the output easier to find. Do we need to make it a pipeline artifact?

Reply:
Yeah, I'll do that on the next PR when I make it part of the pipeline. :)